pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -131,14 +133,21 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tp *v1alpha1.TektonPipel
 	tp.Status.MarkPreReconcilerComplete()
 
 	// Check if an tekton installer set already exists, if not then create
-	existingInstallerSet := tp.Status.GetTektonInstallerSet()
+	labelSelector := fmt.Sprintf("%s=%s,%s=%s",
+		tektoninstallerset.CreatedByKey, createdByValue,
+		tektoninstallerset.InstallerSetType, v1alpha1.PipelineResourceName,
+	)
+	existingInstallerSet, err := tektoninstallerset.CurrentInstallerSetName(ctx, r.operatorClientSet, labelSelector)
+	if err != nil {
+		return err
+	}
 	if existingInstallerSet == "" {
 		createdIs, err := r.createInstallerSet(ctx, tp)
 		if err != nil {
 			return err
 		}
 		// If there was no existing installer set, that means its a new install
-		r.metrics.logMetrics(metricsNew, r.releaseVersion, logger)
+		r.metrics.logMetrics(metricsNew, r.pipelineVersion, logger)
 
 		return r.updateTektonPipelineStatus(ctx, tp, createdIs)
```

Comment:
in `updateTektonPipelineStatus`
we should now skip the `UpdateStatus` api n call 
As it is not necessary to explicitly update now
and let knative/pkg do that at then end of reconciler

Reply:
we could just move this 
```
	tp.Status.SetTektonInstallerSet(createdIs.Name)
	tp.Status.SetVersion(r.pipelineVersion)
```
to the end of main reconciler now

Reply:
we could, but in the current flow, what follows is comparisson of version and targetnamespace, which will be unnecessary right after a new installSet creation.

one goal of this patch was to fix the bug with minimal changes in lifecycle (install/uninstall or metrics). 

So shall we handle all refactoring like this in a different patch.